File: node.js/src/slimmed/domain/books/entities/publisher.ts
import { NonEmptyString, NonEmptyUuid } from '#core/typing';
import { DeepReadonly } from 'ts-essentials';

export type PublisherId = NonEmptyUuid<'PublisherId'>;
export type PublisherName = NonEmptyString<'PublisherName'>;

export type Publisher = DeepReadonly<{ id: PublisherId; name: PublisherName }>;

File: java/src/main/java/io/eventdriven/slimdownaggregates/original/domain/books/entities/Publisher.java
package io.eventdriven.slimdownaggregates.original.domain.books.entities;

import static io.eventdriven.slimdownaggregates.original.infrastructure.validation.Validation.assertNotNull;

public record Publisher(PublisherId id, PublisherName name) {
  public Publisher {
    assertNotNull(id);
    assertNotNull(name);
  }
}
